src/main/java/hwManifestGen/AllComponents.java

```diff
@@ -195,6 +235,276 @@ public static void setOptions(Options options) {
+			while ((inputLine1 = in1.readLine()) != null) {
+			    content1.append(inputLine1);
+			}
+			in1.close();
+
+			// Extract X-Auth-Token from the header of con1
+			String authToken = con1.getHeaderField("X-Auth-Token");
+			System.out.println("X-Auth Token:" + authToken);
+
+			if (type.contains("GET")) {
+			    String urlString2 = "https://" + system + command;
+			// Create URL object
+			URL obj = new URL(urlString2);
+
+			// Create HttpsURLConnection
+			HttpsURLConnection con2 = (HttpsURLConnection) obj.openConnection();
+
+			// Set request method to GET
+			con2.setRequestMethod("GET");
+
+			// Set the X-Auth-Token header
+			con2.setRequestProperty("X-Auth-Token", authToken);
+
+			// Get response code
+			int responseCode = con2.getResponseCode();
+			System.out.println("Response Code: " + responseCode);
+
+			// Read response
+			BufferedReader in = new BufferedReader(new InputStreamReader(con2.getInputStream()));
+			String inputLine;
+			StringBuilder response = new StringBuilder();
+			while ((inputLine = in.readLine()) != null) {
+				response.append(inputLine);
+			}
+			in.close();
+
+			output = response.toString();
+      
+			// Extract the CertificateString
+			String certificateString = "";
+
+			// Find the start index of the CertificateString
+			int startIndex = output.indexOf("CertificateString\":\"") + "CertificateString\":\"".length();
+
+			// Find the end index of the CertificateString
+			int endIndex = output.indexOf("\"", startIndex);
+
+			// Check if valid start and end index are found
+			if (startIndex >= 0 && endIndex >= 0) {
+			    // Extract the CertificateString from the output
+			    certificateString = output.substring(startIndex, endIndex);
+			    
+			    // Replace escape sequence for newline with actual newline character
+			    certificateString = certificateString.replace("\\n", "\n");
+			}
+
+			// Print the extracted CertificateString
+			System.out.println("Certificate String: " + certificateString);
+
+
+			// Set the output file path
+			String file;
+
+			// Check if "path" command option is provided
+			if (cmd.hasOption("path")) {
+			    // Get the value of the "path" command option
+			    String value = cmd.getOptionValue("path");
+			    System.out.println("Value of command option: " + value);
+
+			    // Create the directory if it doesn't exist
+			    File directory = new File(value);
+			    if (!directory.exists()) {
+				boolean created = directory.mkdirs();
+				if (created) {
+				    System.out.println("Directory created: " + value);
+				} else {
+				    System.out.println("Failed to create directory: " + value);
+				}
+			    }
+
+			    // Set the file path to the specified directory with the filename "output2.pem"
+			    file = value + "output2.pem";
+			} else {
+			    System.out.println("Output Path option not provided");
+			    // Set the default output file path
+			    String value = "/home/prerana/Desktop/PCVT/PCVT/src/main/java/hwManifestGen/";
```

Comment:
At the instructions the paths suggested were in general /opt/hpe/scl/ or /opt/hpe/scl/certificates for the outputs... so I suggest to modify this for /opt/hpe/scl/ or a directory whithin this path.

Reply:
Changed to the expected path